providers/jldap/src/java/edu/amc/sakai/user/UserAttributeRoleProvider.java

```diff
@@ -51,6 +47,7 @@ public void init() {
 	public Set<String> getAdditionalRoles(String userId) {
 		if (userId != null) {
 			try {
+				// This should be cached.
```

Comment:
UserDirectoryService.callCache is an existing and highly effective cache of users